gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/PolygonDrawingViewModel.java

```diff
@@ -262,19 +263,48 @@ private boolean isLocationLockEnabled() {
   public void startDrawingFlow(Project selectedProject, Layer selectedLayer) {
     this.selectedLayer.onNext(selectedLayer);
     this.selectedProject.onNext(selectedProject);
-    updateDrawingState(PolygonDrawing.STARTED);
+    polygonDrawingState.onNext(createDrawingState(State.IN_PROGRESS));
```

Comment:
You can expose these calls to `createDrawingState` as a static constructor methods e.g.,`PolygonDrawingState.inProgress()`. Same for other states, passing args that apply to that state.

Reply:
Done